### PR TITLE
pcre: add head, enhance https mirror

### PR DESCRIPTION
**Pull request by vszakats**

### CODE REVIEW

**Comment by vszakats** [comment]
`head` might be particularly useful at this point because it has some [important](https://lists.exim.org/lurker/message/20150529.074548.5d93521a.en.html) [security](https://lists.exim.org/lurker/message/20150601.083857.b2c4cbb7.en.html) [fixes](https://lists.exim.org/lurker/message/20150602.152356.ff2a6251.en.html), which are not yet released under a new version number. Recommended by a major `pcre` contributor [here](https://lists.exim.org/lurker/message/20150606.065758.a932067e.en.html).

File: Library/Formula/pcre.rb [hunk 1]
```diff
@@ -2,9 +2,11 @@ class Pcre < Formula
   desc "Perl compatible regular expressions library"
   homepage "http://www.pcre.org/"
   url "https://ftp.csx.cam.ac.uk/pub/software/programming/pcre/pcre-8.37.tar.bz2"
-  mirror "https://downloads.sourceforge.net/project/pcre/pcre/8.37/pcre-8.37.tar.bz2"
+  mirror "https://www.mirrorservice.org/sites/downloads.sourceforge.net/p/pc/pcre/pcre/8.37/pcre-8.37.tar.bz2"
```
Comment (DomT4) [inline Library/Formula/pcre.rb]: Why this change?

Comment (vszakats) [inline Library/Formula/pcre.rb]: `sf.net` will redirect to an http address. `mirrorservice.org` won't do such thing.

Comment (vszakats) [inline Library/Formula/pcre.rb]: Related: I had left a feature request on sf.net 6+ months ago about SSL/TLS downloads, and this ticket is still in _unread_ status. This seems to indicate a lack of interest on their part to resolve this.

Comment (vszakats) [inline Library/Formula/pcre.rb]: `pcre2` already went [merged](9f746f414f470529e3993de62dfd3ee3b4324fa4) with the same mirror, but I can restore the address, if it is problematic of course.

Comment (DomT4) [inline Library/Formula/pcre.rb]: If Mike already merged it in `pcre2` I've no objections to doing the same.

**Comment by DomT4** [comment]
:rocket: in https://github.com/Homebrew/homebrew/commit/006f0b189358e8fb7981daa31f476633cfe7faae. Thanks @vszakats!
